### DISCUSSION TITLE
Upgrading ethereum provider with Web3Modal v1

### DISCUSSION THREAD

**Discussion by junaama** [WalletConnect/walletconnect-monorepo#2746] [Developer Support]
Hi, 

I saw on the docs that we can update ethereum-provider and keep our Web3Modal v1 implementation. 

> If you still want to use Web3Modal v1.0 but just upgrade the WalletConnect ethereum-provider to v2.0, you can update to the latest version available on NPM which you can find [here](https://npmjs.com/package/@walletconnect/ethereum-provider).

Just to clarify do I only need to install the new ethereum provider package and my current implementation is good to go, or would I need to also switch out using WalletConnectProvider from @walletconnect/web3-provider to EthereumProvider from @walletconnect/ethereum-provider?

Below is my current Web3Modal v1 implementation

```
import WalletConnectProvider from '@walletconnect/web3-provider';
...
...
const web3Modal = new Web3Modal({
  cacheProvider,
  providerOptions: {
    'install-metamask': {
      display: {
        logo: providers.METAMASK.logo,
        name: 'Install MetaMask',
        description: 'Connect using browser wallet',
      },
      package: {},
      connector: async () => {
        window.open('https://metamask.io');
        throw new Error('MetaMask not installed');
      },
    },
    walletconnect: {
      package: WalletConnectProvider
      options: {
        infuraId: process.env.NEXT_PUBLIC_INFURA_ID
      },
    },
  },
});
....
....
```

**Comment by glitch-txs** [comment]
Ethereum-provider is a bit different, I would suggest following its own docs
https://docs.walletconnect.com/2.0/web/providers/ethereum

Also keep in mind, ethereum-provider won't show MetaMask on the modal, for browser wallets to show up you'll need to use the package with wagmi.
